### DISCUSSION TITLE
Flux in ComfyUI

### DISCUSSION THREAD

**Discussion by Enallio** [Comfy-Org/ComfyUI#4335] [Q&A]
Can you give me a hint, please?

Installed ComfyUI. Flux is my first one. 
I downloaded the dev fp16 version, the program works, something is generated, but on the output I have just a black 1024 by 1024 square (or any other resolution I set). 

That said, everything is fine on the fp8 version. What could be the reason? 
I have already tried everything. 

The setup should allow 4080 + 32 RAM

**Comment by ltdrdata** [comment]
You cannot use fp16 4080 due to insufficient vram
Minimum requirement is 4090 or 3090

**Reply by Enallio** [reply]
I'm a complete ignoramus. 

How do people on the internet run this model on the same video card. Is there some kind of a trick to it?

**Reply by 100xlongx** [reply]
they are likely using NF4 @Enallio

**Reply by Enallio** [reply]
> they are likely using NF4 @Enallio
May Be
I changed page file settings. It does not help me at all. 
I repeat some info - I "can" generate with fp16, but all pictures look like black screen.  
![image](https://github.com/user-attachments/assets/c4774ea9-c066-4a5a-b921-9e2c9e6b0464)

*black screed on the right*

**Reply by JorgeR81** [reply]
Try the "normal" scheduler.
That's was I used for fp16.

**Comment by JorgeR81** [comment]
You may need to change your page file settings. 
https://github.com/comfyanonymous/ComfyUI/discussions/4226

I'm on Windows 10 
I can use Flux in fp16, with 32GB RAM and 8GB VRAM ( GTX 1070 ).
I run out of RAM when the models are loading and when I change the prompt. 
But I can generate, because the system uses the page file.

**Reply by Enallio** [reply]
> You may need to change your page file settings. #4226
> 
> I'm on Windows 10 I can use Flux in fp16, with 32GB RAM and 8GB VRAM ( GTX 1070 ). I run out of RAM when the models are loading and when I change the prompt. But I can generate, because the system uses the page file.

It didn't help. Still black screen instead of generating

**Comment by JorgeR81** [comment]
@ltdrdata You can't use Flux, fp16, with your RTX 4080 ?
https://github.com/comfyanonymous/ComfyUI/discussions/2970#discussioncomment-8726370

Maybe it's a Linux issue ?

**Reply by ltdrdata** [reply]
In the case of FLUX fp16, the model size approaches 24GB, while the 4080 has only 16GB VRAM. In this case, to increase VRAM, you need to expand VRAM in the form of shared memory.

The problem is that it's essentially the same as running on CPU, so the generation speed becomes extremely slow.

If you use such settings, even when using a model smaller than 16GB, at some point, part of the model will move to shared memory, similarly causing performance degradation. Therefore, this is a highly discouraged configuration.

**Reply by JorgeR81** [reply]
OK, I see.

With my GTX 1070 ( 8GB VRAM ), fp8 is already so slow, that fp16 it's only about %15 slower ...

Theoretically the NF4 version could fit in 8GB VRAM. Let's hope it's faster.
